File: src/main/java/com/server/ordering/api/CouponApiController.java
package com.server.ordering.api;

import com.server.ordering.domain.dto.ResultDto;
import com.server.ordering.domain.dto.request.CouponDto;
import com.server.ordering.service.CouponService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class CouponApiController {

    private final CouponService couponService;

    /**
     * 쿠폰 등록
     */
    @PostMapping("/api/coupon")
    public ResultDto<Boolean> registerCoupon(@RequestBody CouponDto dto) {
        couponService.saveCoupon(dto);
        return new ResultDto<>(1, true);
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/api/CouponApiController.kt
package com.byeoru.ordering_server.api

import com.byeoru.ordering_server.domain.dto.ResultDto
import com.byeoru.ordering_server.domain.dto.request.CouponDto
import com.byeoru.ordering_server.service.CouponService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController


@RestController
class CouponApiController(val couponService: CouponService) {

    /**
     * 쿠폰 등록
     */
    @PostMapping("/api/coupon")
    fun registerCoupon(@RequestBody dto: CouponDto): ResultDto<Boolean> {
        couponService.saveCoupon(dto)
        return ResultDto(1, true)
    }
}